fhirstore/fhirstore.py

```diff
@@ -206,3 +224,43 @@ def validate(self, resource):
         if schema is None:
             raise Exception(f"missing schema for resource {resource}")
         validate(instance=resource, schema=schema)
+
+    def search(self, resourceType, params):
+        """
+        Searchs for params inside a resource.
+        Returns a bundle of items, as required by FHIR standards.
+
+        Args:
+            - resource: FHIR resource (eg: 'Patient')
+            - params: search parameters as returned by the API. For a simple
+            search, the parameters should be of the type {"key": "value"}
+            eg: {"gender":"female"}, with modifiers {"address.city:exact":"Paris"}.
+            For a search on multiple parameters, params should be a payload of the form:
+            {"key1": ["value1", "value2"], "multiple": {"key2": ["value3", "value4"]}}.
+        Returns: A bundle with the results of the search, as required by FHIR
+        search standard.
+        """
+        res = []
+        
+        if len(params) == 0:
+            query = {"query": {"match_all": {}}}
+        elif len(params) == 1 and not params["multiple"]:
+            query = dict()
+            query["min_score"] = 0.01
+            query["query"] = element_search(params)
+        else:
+            if params["multiple"]:
```

Comment:
I'm not sure we need multiple levels of if/else (it makes it difficult to read)
also there is an issue with "multiple" as discussed together